internal/simplecue/utils.go

```diff
@@ -272,3 +276,31 @@ func hasStructFields(v cue.Value) bool {
 
 	return false
 }
+
+func areCuePathsFromSameRoot(a cue.Path, b cue.Path) bool {
+	selectorsA := a.Selectors()
+	selectorsB := b.Selectors()
+
+	if len(selectorsA) == 0 || len(selectorsB) == 0 {
+		return false
+	}
+
+	return selectorsA[0].String() == selectorsB[0].String()
+}
+
+func cuePathIsChildOf(root cue.Path, maybeChild cue.Path) bool {
+	selectorsRoot := root.Selectors()
+	selectorsChild := maybeChild.Selectors()
+
+	if len(selectorsRoot) > len(selectorsChild) {
```

Comment:
Could happen that child has more selectors than the root?

Reply:
Yeah :)

```cue
schema: {
  spec: { // path: schema.spec (2 selectors)
    title: string
    origin: #Origin
  }
  #Container: {
    foo: string
    #OtherContainer: {
      bar: string

      #Origin: { creator: string } // path: schema.#Container.#OtherContainer.#Origin (4 selectors)
    }
  }
}
```

Reply:
And in the next loop you are checking if any field is different to return false, but if `selectorsChild` is bigger, you aren't checking all the fields, right? Or am I missing something?

Reply:
Yeah, the idea is to check if one selector is a "subset" of the other.

Let's imagine:
* `root = schema.spec`
* `maybeChild = schema.spec.#Container.#Origin`
* in that case, we are not in an "out of root" scenario as `root` is a subset of `maybeChild`

Whereas in this example, we do have an "out of root" reference:
* `root = schema.spec`
* `maybeChild = schema.#Container.#Origin`

In both cases, we need to check the selectors of `root` to make a decision but we might not have to check all the selectors of `maybeChild`